src/AnnotationDialog.js

```diff
@@ -231,7 +238,8 @@ class AnnotationDialog extends EventEmitter {
             this.hasAnnotations = true;
         }
 
-        this.addAnnotationElement(annotation);
+        this.annotations.push(annotation);
```

Comment:
Not exactly a comment on this line, but would it make sense to get rid of `this.hasAnnotations` if we can now derive that from `this.annotations.length`?

Reply:
I can change hasAnnotations into a util method because it requires checking not only the length of the annotations array but if the only annotation in the array is a plain highlight or not as the behavior differs on that as well

Reply:
Sorry... confused this w/ hasComments which is another terrible boolean. Yes I can remove hasAnnotations :)